src/test/java/nl/thehyve/ocdu/validators/PatientDataOcChecksTests.java

```diff
@@ -0,0 +1,183 @@
+
+        //date of enrollment should be in the past
+        subject.setDateOfEnrollment("01-JUN-3012");
+        DateOfEntrollmentPatientDataCheck check = new DateOfEntrollmentPatientDataCheck();
+        ValidationErrorMessage error = check.getCorrespondingError(0, subject, metadata);
+        assertThat(error.getMessage(), containsString("past"));
+    }
+
+    @Test
+    public void testDateOfEnrollmentInvalidFormat() {
+        Subject subject = new Subject();
+        subject.setSsid("1234");
+
+        //invalid date format
+        subject.setDateOfEnrollment("01-JU");
+        DateOfEntrollmentPatientDataCheck check = new DateOfEntrollmentPatientDataCheck();
+        ValidationErrorMessage error = check.getCorrespondingError(0, subject, metadata);
+        assertThat(error.getMessage(), containsString("invalid"));
+    }
+
+    @Test
+    public void testPersonIdProvided() {
+        Subject subject = new Subject();
+        subject.setSsid("1234");
+
+        //person id is provided
+        subject.setPersonId("1357");
+        PersonIdPatientDataCheck check = new PersonIdPatientDataCheck();
+        ValidationErrorMessage error = check.getCorrespondingError(0, subject, metadata);
+        assertThat(error.getMessage(), containsString("Person"));
+    }
+
+
+    @Test
+    public void testSecondaryIdTooLong() {
+        Subject subject = new Subject();
+        subject.setSsid("1234");
+
+        //secondary id is provided, but too long
+        subject.setSecondaryId("1111112222222222333333333444444444445555555555666666666667777777777888888");
+        SecondaryIdPatientDataCheck check = new SecondaryIdPatientDataCheck();
+        ValidationErrorMessage error = check.getCorrespondingError(0, subject, metadata);
+        assertThat(error.getMessage(), containsString("length"));
+    }
+
+    @Test
+    public void testStudyNotProvided() {
+        Subject subject = new Subject();
+        subject.setSsid("1234");
+
+        //study is not provided
+        subject.setStudy("");
+        StudyPatientDataCheck check = new StudyPatientDataCheck();
+        ValidationErrorMessage error = check.getCorrespondingError(0, subject, metadata);
+        assertThat(error.getMessage(), containsString("Study should"));
+    }
+
+    @Test
+    public void testSitesNotExist() {
+        Subject subject = new Subject();
+        subject.setSsid("1234");
+
+        //sites do not exist
+        subject.setStudy("S_STUDY1");
+        subject.setSite("myownsitethatdoesnotexist");
+        SitePatientDataCheck check = new SitePatientDataCheck();
+        ValidationErrorMessage error = check.getCorrespondingError(0, subject, metadata);
+        assertThat(error.getMessage(), containsString("exist"));
+    }
+
+    @Test
+    public void testSiteExistSuccess() {
+        Subject subject = new Subject();
+        subject.setSsid("1234");
+
+        //sites do not exist
+        subject.setStudy("S_STUDY1");
+        subject.setSite("Sjogren - Sjogren");
```

Comment:
okay Jacob confirmed that we need indeed this Identifier as it is - it is a technical ID of the site we will need for making a SOAP call.